### PR TITLE
Post 2.3.0~alpha1 release bumps and fixes

### PR DESCRIPTION
**Pull request by kit-ty-kate**
The release process for 2.3.0~alpha1 was pretty eventful and a couple of fixes were required after the merge of https://github.com/ocaml/opam/pull/5381 which we missed.

This PR is also bumping master to 2.4.0~alpha1 as a new 2.3 branch was already created. ~I'll create a similar PR for the 2.3 branch as well~

Backported in https://github.com/ocaml/opam/pull/6205

### CODE REVIEW

File: release/Dockerfile.in [hunk 1]
```diff
@@ -27,6 +27,7 @@ CMD { tar xz && \
       ./configure --with-vendored-deps --with-mccs && \
       echo "(${LINKING})" > src/client/linking.sexp && \
       make opam && \
+      chmod u+w _build/default/src/client/opamMain.exe && \
       strip opam ; \
     } >&2 && \
     cat opam
```
Comment (dra27) [inline release/Dockerfile.in:33]: Wouldn't it be slightly better to copy it here? (or even potentially move this into `Makefile` - `make opam-stripped` or some such)? Along the lines of:
```suggestion
```

```suggestion
      cp opam opam-stripped && \
      chmod 775 opam-stripped && \
      strip opam-stripped ; \
    } >&2 && \
    cat opam-stripped
```

File: release/Makefile [hunk 2]
```diff
@@ -124,6 +123,7 @@ host: $(OUTDIR)/opam-full-$(VERSION).tar.gz build/$(HOST).env
 	  echo "$(call LINKING,$(HOST_OS))" >src/client/linking.sexp && \
 	  $(MAKE) opam; \
 	)
+	chmod u+w build/opam-full-$(VERSION)/_build/default/src/client/opamMain.exe
```
Comment (dra27) [inline release/Makefile:126]: Similarly here: in fact, the duplication kinda increases the case for having `$(MAKE) opam-stripped`?

**Review by dra27 — APPROVED** [review]
All good - on a combined altar of removing some Dune magic from the scripts and reducing code duplication, it looks like it would be worth adding a target to `Makefile` which produces a stripped opam executable.

File: release/Makefile (hunk 2 above)
Comment (kit-ty-kate) [reply release/Makefile:126]: done in https://github.com/ocaml/opam/pull/6208

**Comment by avsm** [comment]
Just a minor data point that makes it a bit easier to package in operating systems is to use similar targets to GNU projects: in this case, it would be `install-strip`: https://www.gnu.org/prep/standards/html_node/Standard-Targets.html

**Comment by kit-ty-kate** [comment]
in our case the stripped binary is not installed but only copied to the root directory of the project

**Pull request merged**